### PR TITLE
feat(native-filters): enable filter indicator and make datasource optional

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY
This PR decouple decouple filter params from native filters config modal. In this phase we decoupled datasource based on `datasourceCount` property that exported from every viz-plugin

More changes:
- we update filter badge logic for a case if filter doesn't have datasource, in this case we show applied indicator if this chart affected by scope
- we build filter types dropdown in filter config modal based on behavior property exported from viz-plugin

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/56388545/108049050-748ca100-7050-11eb-91c9-9b6c47551894.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/filters/components/Range/index.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ import thumbnail from './images/thumbnail.png';
 export default class AntdRangeFilterPlugin extends ChartPlugin {
   constructor() {
     const metadata = new ChartMetadata({
-      name: t('Range filter plugin'),
+      name: t('Range filter'),
```
Comment (villebro) [inline superset-frontend/src/filters/components/Range/index.ts:28]: Slightly unrelated, but while we're at it I think we should drop the `Antd` prefix from the class names and just go with `RangeFilterPlugin` (same for the others).

**Review by villebro — APPROVED** [review]
Wow, this is incredible once again! Nothing much to add here, except perhaps the renaming of the filter classes and changing the title from refactor to feat, as this is also adding significant new functionality.

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterValue.tsx [hunk 2]
```diff
@@ -51,15 +51,19 @@ const FilterValue: React.FC<FilterProps> = ({
...
   const filterState = useFilterState(id);
-  const [loading, setLoading] = useState<boolean>(true);
   const [state, setState] = useState([]);
   const [error, setError] = useState<boolean>(false);
   const [formData, setFormData] = useState<Partial<QueryFormData>>({});
   const inputRef = useRef<HTMLInputElement>(null);
   const [target] = targets;
-  const { datasetId = 18, column } = target;
+  const {
+    datasetId = 18,
```
Comment (amitmiran137) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterValue.tsx:60]: why 18?

Comment (simcha90) [reply superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterValue.tsx:60]: no idea, it was here

Comment (simcha90) [reply superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterValue.tsx:60]: but may be good remove it

**Pull request merged**